va/va.c

```diff
@@ -719,15 +661,78 @@ VAStatus vaSetDriverName(
     return VA_STATUS_SUCCESS;
 }
 
+static VAStatus va_new_opendriver(VADisplay dpy)
+{
+    VADisplayContextP pDisplayContext = (VADisplayContextP)dpy;
+    /* In the extreme case we can get up-to 5ish names. Pad that out to be on
+     * the safe side. In the worst case, the DRIVER BUG below will print and
+     * we'll be capped to the current selection.
+     */
+    char *drivers[20] = { 0, };
+    unsigned int num_drivers = ARRAY_SIZE(drivers);
+    VAStatus vaStatus;
+    const char *driver_name_env;
+    VADriverContextP ctx;
+
+    /* XXX: Temporary dummy return, until all platforms are converted */
+    if (!pDisplayContext->vaGetDriverNames)
+        return VA_STATUS_ERROR_INVALID_PARAMETER;
+
+    /* XXX: The order is bonkers - env var should take highest priority, then
+     * override (which ought to be nuked) than native. It's not possible atm,
+     * since the DPY connect/init happens during the GetDriverNames.
+     */
+    vaStatus = pDisplayContext->vaGetDriverNames(pDisplayContext, drivers, &num_drivers);
+    if (vaStatus != VA_STATUS_SUCCESS) {
+        /* Print and error yet continue, as per the above ordering note */
+        va_errorMessage(dpy, "vaGetDriverNames() failed with %s\n", vaErrorStr(vaStatus));
+    } else if (num_drivers > ARRAY_SIZE(drivers)) {
+        va_errorMessage(dpy, "DRIVER BUG: vaGetDriverNames() provides too many drivers\n");
+        num_drivers = ARRAY_SIZE(drivers);
+    }
+
+    ctx = CTX(dpy);
+    driver_name_env = secure_getenv("LIBVA_DRIVER_NAME");
+
+    if ((ctx->override_driver_name) || (driver_name_env && (geteuid() == getuid()))) {
+        const char *driver = ctx->override_driver_name ?
+                             ctx->override_driver_name : driver_name_env;
+
+        drivers[0] = strdup(driver);
+        num_drivers = 1;
```

Comment:
also , maybe has the second driver name, so driver[1] may leak

Reply:
Indeed we are leaking all drivers - quite busy ATM, will open a MR in a few days unless someone beats me to it.

Note that as the XXX comment just above says, the current ordering is off. So there is a serious work to be done here and fixing things.

Reply:
Should be fixed with https://github.com/intel/libva/pull/719 ... plus a typo fix.